service/migration_manager.cc

```diff
@@ -1215,7 +1215,9 @@ future<column_mapping> get_column_mapping(table_id table_id, table_schema_versio
 }
 
 future<> migration_manager::on_join(gms::inet_address endpoint, gms::endpoint_state ep_state) {
-    schedule_schema_pull(endpoint, ep_state);
+    if (!_group0_client.using_raft()) {
```

Comment:
In 605e53e617812d9c26a24927167f23718cc8b396, a change was made which can make using_raft() return false for a brief time during boot. To really avoid schema pulls, we would either have to make sure that no gossiper events fire at this time (probably not viable), or have a stronger using_raft() which is monotonic and never reverts to false temporarily. Here, we can use a stronger version because we don't make any action in case using_raft() is true.

@gleb-cloudius @kbr-scylla

Reply:
It's not a big deal if there are false positives during the transition period. Going forward we will delete this code. E.g. right after making raft mandatory it can be deleted. I think we're overthinking it.

Reply:
It's going to be a problem with topology on raft, because schema depends on topology (new table on tablets). If we don't use raft, we may get schema agreement but requests will be failing. We can delay fixing this, but we have to make sure it's fixed before topology on raft is production ready.

Reply:
But without the patch we always pull which is worse, so the patch makes the state better. And since it uses `using_raft()`  which hides the workaround the code will start working correctly when the problem will be fixed.
I am willing to look into it as part of QA effort.

Reply:
That depends on how you implement the fix. Here the pulls are initiated by the booting node in reaction to gossip, not in reaction to other nodes sending requests, so the fix for the latter may not impact this code.

It's fine to improve incrementally, but if we're leaving holes behind the PR should not have "Fixes #12870".